### DISCUSSION TITLE
CRTK interface using different frames

### DISCUSSION THREAD

**Discussion by micsc12** [surgical-robotics-ai/surgical_robotics_challenge#32] [Q&A] [answered]
Hi. 

I'm trying to use the CRTK interface to control steer the PSM towards the needle pose. However, im a bit confused by how the frames have been setup, namely that some topics publish with frame_id: World, and some use frame_id=baselink. 

Ie. /CRTK/psm2/measured_cp uses "baselink" frame, where as 
CRTK/Entry1/measured_cp and CRTK/Needle/measured_cp uses world frame. 

Now, to convert the transforms given in baselink to world should be trivial, but i cant seem to find the correct baselink transform. Ie. /ambf/env/psm2/baselink/State reports the base link to be at (-1, 1.346, 1,349) but using transform to convert  /CRTK/psm2/measured_cp to World coordinates doesn't seem to give me the right results. 

Therefore, can you provide some more insights on how the frame structure is set up in the simulator, and maybe how to convert between the different frames? 

Thanks in advance :)

//Michael

**Comment by micsc12** [comment]
I stumbled on [this](https://github.com/collaborative-robotics/surgical_robotics_challenge/blob/master/docs/scene_coordinate_frames.md) document, and calling get_T_w_b() on the PSM class works. However, it would be nice to have this available directly as a rostopic thorugh the CRTK interface :)

**Reply by adnanmunawar** [reply]
Hi Michael, thanks for bringing this up. I agree that accessing T_b_w() should be possible through the CRTK interface, I shall add it soon :).

**Comment by adnanmunawar** [comment, marked as answer]
Hi Michael, 
From this [commit](https://github.com/collaborative-robotics/surgical_robotics_challenge/commit/47af8a9eed0f1d528e968e90ac11c9d3a9190f4a) onwards, `T_b_w` for the PSMs should now accessible via the topics `/CRTK/psm1/T_b_w` and `/CRTK/psm2/T_b_w`.

**Reply by micsc12** [reply]
Thanks for this :) Simplifies things a lot :)
